### PR TITLE
added devise

### PR DESCRIPTION
**Pull request by humayunk**
- went through slides for devise set-up
- still need to update user model with appropriate fields

### CODE REVIEW

File: app/controllers/pages_controller.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 class PagesController < ApplicationController
+  skip_before_action :authenticate_user!, only: :home
```
Comment (VaedaSolomon) [inline app/controllers/pages_controller.rb:2]: add :show (as show page is available to view before authentication)

File: db/migrate/20240527204430_devise_create_users.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+class DeviseCreateUsers < ActiveRecord::Migration[7.1]
+  def change
+    create_table :users do |t|
+      ## Database authenticatable
+      t.string :email,              null: false, default: ""
```
Comment (VaedaSolomon) [inline db/migrate/20240527204430_devise_create_users.rb:7]: require another migration for adding columns to User model: first_name, last_name, age etc

**Review by VaedaSolomon — CHANGES_REQUESTED** [review]
Reviewed and provided 2 comments

File: db/migrate/20240527204430_devise_create_users.rb (hunk 2 above)
Comment (humayunk) [reply db/migrate/20240527204430_devise_create_users.rb:7]: added

File: app/controllers/pages_controller.rb (hunk 1 above)
Comment (humayunk) [reply app/controllers/pages_controller.rb]: added

**Review by VaedaSolomon — APPROVED** [review]
reviewed and approved

**Pull request merged**